test/extended/oauth/sampleoauthserver.go

```diff
@@ -74,15 +58,28 @@ var _ = g.Describe("[Suite:openshift/oauth/run-oauth-server] Run the integrated
 		o.Expect(err).ToNot(o.HaveOccurred())
 		e2e.Logf("got the OAuth server address: %s", serverAddress)
 
-		http.DefaultTransport.(*http.Transport).TLSClientConfig = &tls.Config{InsecureSkipVerify: true} // FIXME: VERY VERY UGLY, don't do this at home
+		tlsClientConfig, err := rest.TLSConfigFor(oc.AdminConfig())
+		o.Expect(err).NotTo(o.HaveOccurred())
+		http.DefaultTransport.(*http.Transport).TLSClientConfig = tlsClientConfig
 		resp, err := http.Get(serverAddress)
 		o.Expect(err).ToNot(o.HaveOccurred())
 		defer resp.Body.Close()
-
-		time.Sleep(2 * time.Minute)
 		body, err := ioutil.ReadAll(resp.Body)
 		e2e.Logf("The body received: %s", string(body))
 		o.Expect(err).ToNot(o.HaveOccurred())
+
+		// I could not quite figure this out
```

Comment:
I don't think the following code even does requests to the oauth server you configured :thinking: I'll try to see what can be done here.

Reply:
Hm, nvm, the OAuth server just updates the challenging client to point to it. That's bad with regards to the parallel running of the tests. Crap.

Reply:
It might be possible to work around this by backing the `openshift-challenging-client` up before deploying our own oauth server (needs to be done with global lock to avoid races), restore the original values once the oauth server bootstrap finished, create our own clients with proper values and use those instead.

I'll try to get this working first, and then see what we can do about the above. This is quite a setback though, makes me sad.